### DISCUSSION TITLE
filtrar no dart

### DISCUSSION THREAD

**Discussion by Dablinlabs** [Flutterando/forum#41] [Perguntas e Respostas] [answered]
salve galera boa tarde, é uma duvida bem de iniciante mas estou tendo um problema, no codigo abaixo eu simulei o problema que estou tendo.
Eu preciso filtrar dentro da lista obj todos os myobj que tenham algum dos times de filtersTeams

```dart
class myObj {
  final String name;
  final List<String> team;
  
  myObj ({
    required this.name,
    required this.team,
  });
}

void main() {
 List<myObj> obj = [
    myObj(name: "Name 1", team: ['1', '2']),
    myObj(name: "Name 2", team: ['2']),
    myObj(name: "Name 3", team: ['3']),
  ];
  
  List<String> filtersTeams = ['1', '3'];
  
  List<myObj> filteredList = obj      // filtro nao esta funcionando
      .where((e) => filtersTeams.contains(e.team))
      .toList(); 
  
  filteredList.forEach((e) => print(e.name));
}

```

eu quero salvar dentro da minha lista filteredList todos os objetos que no atributo team tenha pelo menos 1 dos itens que tem dentro de filtersTeams

no caso teria que ter salvar os seguintes dados na filteredList
myObj(name: "Name 1", team: ['1', '2']),
myObj(name: "Name 3", team: ['3']),

**Comment by jacobaraujo7** [comment]
Olá.
Não está muito claro que tipo de problema vc está tendo.

**Reply by Dablinlabs** [reply]
opa perdao acho que ficou meio confuso
eu quero salvar dentro da minha lista filteredList todos os objetos que no atributo team tenha pelo menos 1 dos itens que tem dentro de filtersTeams

**Reply by Dablinlabs** [reply]
no caso teria que ter os seguintes dados na filteredList 
myObj(name: "Name 1", team: ['1', '2']),
myObj(name: "Name 3", team: ['3']),

**Reply by fredfv** [reply]
Tenta utilizar a comparação com o .any ao invez do contains

**Comment by eduardoflorence** [comment, marked as answer]
Olá @Lanysc, você pode fazer assim:

```dart
class myObj {
  final String name;
  final List<String> team;
  
  myObj ({
    required this.name,
    required this.team,
  });
}

void main() {
 List<myObj> obj = [
    myObj(name: "Name 1", team: ['1', '2']),
    myObj(name: "Name 2", team: ['2']),
    myObj(name: "Name 3", team: ['3']),
  ];
  
  List<String> filtersTeams = ['1', '3'];
  
  List<myObj> filteredList = [];
  
  for (var item in obj) {
    if (item.team.any((team) => filtersTeams.contains(team))) {
      filteredList.add(item);
    }
  }
 
  filteredList.forEach((e) => print(e.name));
}
```

O seu exemplo não está dando certo, pois `contains` verifica se uma lista contém um elemento e não uma outra lista.

**Reply by Zeca-dev** [reply]
Você pode, em vez de filtrar filteredTeams, filtrar e.teams. Ou seja, trazer os objetos que tem dentro de Teams, os elementos da filteredTeams.
